infrastructures/openshift/src/main/java/org/eclipse/che/workspace/infrastructure/openshift/OpenshiftMachine.java

```diff
@@ -67,17 +84,117 @@ public String getName() {
 
     @Override
     public Map<String, ? extends Server> getServers() {
-        //TODO https://github.com/eclipse/che/issues/5687
-        return new HashMap<>();
+        //TODO https://github.com/eclipse/che/issues/5688
+        try (OpenShiftClient client = clientFactory.create()) {
+            List<Service> services = client.services()
```

Comment:
It seems that services are used only to get _matchedServices_, so maybe it's better to get rid of the codeblock below and use stream + filter to get _matchedServices_ here?